### PR TITLE
Generate `target` roll options before extracting `EphemeralEffect`s

### PR DESCRIPTION
**Pull request by In3luki**
This came up while I was testing the combination of a `MarkToken` with an `EphermeralEffect` predicated on the mark.
Fully replacing the original roll options with the new options from the clone probably is the safest way to do it.

Test Feat (Mastemind Racket):
```json
{
  "key": "EphemeralEffect",
  "predicate": [
    "target:mark:effect-mastermind"
  ],
  "selectors": [
    "attack",
    "strike-damage"
  ],
  "uuid": "Compendium.pf2e.conditionitems.AJh5ex99aV6VTggg"
}
```
```json
{
  "key": "MarkToken",
  "slug": "effect-mastermind"
}
```

### CODE REVIEW

**Comment by stwlam** [comment]
What's the "this" that came up?

File: src/module/actor/base.ts [hunk 1]
```diff
@@ -798,14 +798,25 @@ class ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
...
 
+        // Target roll options
+        const setBaseTargetOptions = (targetOptions: string[]): void => {
+            if (targetToken && targetOptions.length > 0) {
+                targetOptions.push("target"); // An indicator that there is a target of any kind
+                const mark = this.synthetics.targetMarks.get(targetToken.document.uuid);
+                if (mark) targetOptions.push(`target:mark:${mark}`);
+            }
+        };
+        let targetOptions = targetToken?.actor?.getSelfRollOptions("target") ?? [];
```
Comment (stwlam) [inline src/module/actor/base.ts:809]: No `let`

**Comment by In3luki** [comment]
> What's the "this" that came up?

Oh, right. The `target` predicate doesn't work because the roll option is set after the ephemeral effect is already resolved.

File: src/module/actor/base.ts (hunk 1 above)
Comment (In3luki) [reply src/module/actor/base.ts]: Done

File: src/module/actor/base.ts [hunk 2]
```diff
@@ -798,14 +798,26 @@ class ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
                 ? [`origin:distance:${distance}`, `target:distance:${distance}`]
                 : [null, null];
 
+        // Target roll options
+        const getTargetRollOptions = (actor?: ActorPF2e | null): string[] => {
```
Comment (stwlam) [inline src/module/actor/base.ts:802]:
```suggestion
        const getTargetRollOptions = (actor: Maybe<ActorPF2e>): string[] => {
```

File: src/module/actor/base.ts [hunk 3]
```diff
@@ -798,14 +798,26 @@ class ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
                 ? [`origin:distance:${distance}`, `target:distance:${distance}`]
                 : [null, null];
 
+        // Target roll options
+        const getTargetRollOptions = (actor: Maybe<ActorPF2e>): string[] => {
+            const targetOptions = actor?.getSelfRollOptions("target") ?? [];
+            if (targetToken && targetOptions.length > 0) {
```
Comment (stwlam) [inline src/module/actor/base.ts:804]:
```suggestion
            if (targetToken) {
```

**Pull request merged**